roles/repository/tasks/main.yml

```diff
@@ -41,6 +41,25 @@
           JAVA_TOOL_OPTIONS: "{{ temp_array }}"
       when: temp_array|length > 0
 
+- name: Install 'curl' on RedHat 8 & Redhat 9
+  become: true
+  ansible.builtin.dnf:
+    name: curl
+    state: present
+    allowerasing: true
+  when:
+    - ansible_os_family == 'RedHat'
+    - ansible_distribution_major_version in ['8', '9']
+
+- name: Install 'curl' on RedHat 7
+  become: true
+  ansible.builtin.yum:
+    name: curl
+    state: present
+  when:
+    - ansible_os_family == 'RedHat'
+    - ansible_distribution_major_version == '7'
+
```

Comment:
seems too complex - the point is that on recent RHEL a curl-minimal [began to ship into base image](https://fedoraproject.org/wiki/Changes/CurlMinimal_as_Default#Summary) because it's a requirement for dnf and it's fine that we just use curl-minimal until there are clear issues with it.

having a task which use the generic package module to install `curl` everywhere except on redhat >= 9 should be fine